desktop/core/src/desktop/js/reactComponents/FileChooser/FileChooserModal/FileChooserModal.tsx

```diff
@@ -1,33 +1,100 @@
-import React from 'react';
+// Licensed to Cloudera, Inc. under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  Cloudera, Inc. licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import React, { useState, useEffect } from 'react';
 import Modal from 'antd/lib/modal/Modal';
+import { Menu, Spin } from 'antd';
+
+import HdfsIcon from '../../../components/icons/HdfsIcon';
+import S3Icon from '../../../components/icons/S3Icon';
+import AdlsIcon from '../../../components/icons/AdlsIcon';
 
+import { fetchFileSystems } from '../api';
+import { FileSystem } from '../types';
+import './FileChooserModal.scss';
 interface FileProps {
   show: boolean;
   onCancel: () => void;
-  title: string;
-  okText: string;
+  title?: string;
+  okText?: string;
 }
 
 const defaultProps = { title: 'Choose a file', okText: 'Select' };
 
 const FileChooserModal: React.FC<FileProps> = ({ show, onCancel, title, okText }) => {
+  const [fileSystemList, setFileSystemList] = useState<FileSystem[] | undefined>();
+  const [loading, setLoading] = useState(true);
+
+  const icons = {
+    hdfs: <HdfsIcon />,
+    abfs: <AdlsIcon />,
+    s3: <S3Icon />
+  };
+
   const handleOk = () => {
-    //temporary until the file is selected through the file chooser compoent
+    //temporary until the file is selected through the file chooser component
     onCancel();
   };
 
   const handleCancel = () => {
     onCancel();
   };
 
+  const onPathChange = e => {
+    //TODO: Use fileSystemList[e.key].user_home_dir to retrieve files
+  };
+
+  useEffect(() => {
+    if (show && !fileSystemList) {
```

Comment:
Unless getting the fileSystemsObj API call is very expensive I think it makes more sense to load this before we open the modal (e.g. in the FileChooserWithButton.tsx). Now we are forcing the user to watch a spinner once the modal has been open...

Reply:
The option is to load on click which would delay the opening of the modal itself until we have the data, and would require having a spinner in the button. My take is that the API call will be pretty instant in most cases so the user wouldn't even notice it.

If we have it in the modal it also means that the logic is contained in that module, a cleaner SoC. Instead of having FileChooserWithButton we could use a regular Button. We should also consider that there might be cases where you don't want to trigger the modal from a button.

Reply:
My idea was to retrieve the data when the FileChooserWithButton (or a standard button if desired) is first rendered, not when clicked, so that the filebrowser is directly ready when the user needs it. We could still have one module that exports the modal, the FileChooserWithButton and the data loading functionality, with the SoC even stricter where the actual modal is only concerned with the visual stuff and not the backend stuff.

Anyway, I think this is a bigger discussion around how do design components and where to load data and I might be coloured by working too long on a reusable component library where these things are of greater concern.

Reply:
But it won't be directly ready unless we also fetch the file listing in the button. Which to me feels a bit wasteful in the case where the user doesn't click the button at all..

Reply:
Ok, wasn't sure if more calls where needed or if it would be expensive. If it becomes costly we shouldn't do it.